src/map/mod.rs

```diff
@@ -74,116 +140,128 @@ where
     }
 }
 
-impl<State, Action, C> ViewMarker for Map<State, Action, C>
-where
-    C: MapChild<State, Action>,
-    C::ViewState: MapChildViewState,
-{
-}
+impl<State, Action, Children> ViewMarker for Map<State, Action, Children> {}
 
-pub struct MapViewState<State, Action, C>
-where
-    C: MapChild<State, Action>,
-    C::ViewState: MapChildViewState,
-{
+pub struct MapViewState<DS, CS> {
+    map_dom_state: DS,
+    children_state: CS,
     leaflet_map: leaflet::Map,
-    children: Vec<(C::Element, C::ViewState)>,
 }
 
-#[derive(Debug)]
-enum MapMessage {
+#[derive(Debug, Clone, Copy)]
+pub enum MapMessage {
     InitMap,
 }
 
-impl<State, Action, C> View<State, Action, ViewCtx, DynMessage> for Map<State, Action, C>
+/// Distinctive id for better debugging
+const MAP_VIEW_ID: ViewId = ViewId::new(1236068);
+
+impl<State, Action, Children> View<State, Action, ViewCtx, DynMessage>
+    for Map<State, Action, Children>
 where
     State: 'static,
     Action: 'static,
-    C: MapChild<State, Action>,
-    C::ViewState: MapChildViewState,
+    Children: MapChildren<State>,
 {
-    type Element = Pod<web_sys::HtmlElement, ElementProps>;
+    type Element = <MapDomView<State, Action> as View<State, Action, ViewCtx, DynMessage>>::Element;
 
-    type ViewState = MapViewState<State, Action, C>;
+    type ViewState = MapViewState<
+        <MapDomView<State, Action> as View<State, Action, ViewCtx, DynMessage>>::ViewState,
+        Children::SeqState,
+    >;
 
     fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
-        // TODO:
-        // Is there a way to just use `html::div(())`?
-        let mut pod = Pod::new_element(vec![], HTML_NS, "div");
-        pod.props
-            .styles()
-            .set_style("width".into(), Some("100%".into()));
-        pod.props
-            .styles()
-            .set_style("height".into(), Some("100%".into()));
-        pod.node.apply_props(&mut pod.props);
-
-        let map_options = leaflet::MapOptions::default();
-        let html_el = pod.node.unchecked_ref::<web_sys::HtmlElement>();
-        let leaflet_map = leaflet::Map::new_with_element(html_el, &map_options);
-
-        let children = self
-            .children
-            .iter()
-            .map(|c| c.build(ctx))
-            .collect::<Vec<_>>();
-
-        let view_state = MapViewState {
-            leaflet_map,
-            children,
-        };
-
-        // We have to postpone the map initiation
-        // because the DOM element has been created at this point in time
-        // but has not yet been mounted.
-        let thunk = ctx.message_thunk();
-        spawn_local(async move {
-            thunk.push_message(MapMessage::InitMap);
-        });
-        (pod.into(), view_state)
+        ctx.with_id(MAP_VIEW_ID, |ctx| {
```

Comment:
Hmm when the message type stays the same (`MapMessage`) within the children, I don't think this is actually necessary.

Reply:
Actually, thinking about it, I think it is at least safer.
E.g. when a direct child would receive a Message, it would be ambiguous whether the message is meant for the parent or the child